alphacsc/tests/test_solver_d.py

```diff
@@ -0,0 +1,269 @@
+import pytest
+from numpy.testing import assert_allclose
+
+import numpy as np
+
+from alphacsc.tests.conftest import N_TRIALS, N_CHANNELS, N_TIMES_ATOM, N_ATOMS
+
+from alphacsc.loss_and_gradient import compute_objective
+from alphacsc.utils import check_random_state, construct_X_multi
+from alphacsc.update_d_multi import prox_d, prox_uv
+from alphacsc._solver_d import get_solver_d
+
+
+@pytest.fixture
+def D_init(rng, shape):
+    return rng.randn(*shape)
+
+
+@pytest.mark.parametrize('rank1, solver_d, uv_constraint', [
+    (False, 'auto', 'auto'),
+    (False, 'fista', 'auto'),
+    (True, 'joint', 'joint'),
+    (True, 'fista', 'joint'),
+    (True, 'alternate_adaptive', 'separate'),
+    (True, 'alternate', 'separate'),
+    (True, 'auto', 'separate')
+])
+@pytest.mark.parametrize('window', [True, False])
+@pytest.mark.parametrize('momentum', [True, False])
+def test_get_solver_d(rank1, solver_d, uv_constraint, window, momentum):
+    """Tests valid values."""
+
+    d_solver = get_solver_d(solver_d=solver_d,
+                            rank1=rank1,
+                            window=window,
+                            momentum=momentum)
+
+    assert d_solver is not None
+
+
+@pytest.mark.parametrize('solver_d', ['alternate', 'alternate_adaptive',
+                                      'joint'])
+def test_get_solver_d_error_solver(solver_d):
+    """Tests for the case rank1 is False and solver_d is not compatible."""
+
+    with pytest.raises(AssertionError,
+                       match="solver_d should be auto or fista. Got*"):
+
+        get_solver_d(solver_d=solver_d,
+                     rank1=False,
+                     window=True,
+                     momentum=False)
+
+
+@pytest.mark.parametrize('uv_constraint', ['separate', 'joint'])
+def test_get_solver_d_error_uv_constraint(uv_constraint):
+    """Tests for the case rank1 is False and uv_constraint is not
+    compatible."""
+
+    with pytest.raises(AssertionError,
+                       match="solver_d should be auto or fista. Got*"):
```

Comment:
It does not seem to be the right error message no? Should be something about bad `uv_constraint`?